Geometry/HGCalCommonData/src/HGCalWafer.cc

```diff
@@ -0,0 +1,221 @@
+#include "Geometry/HGCalCommonData/interface/HGCalWafer.h"
+#include <vector>
+
+HGCalWafer::HGCalWafer(double waferSize, int32_t nFine, int32_t nCoarse) {
+  N_[0] = nFine;
+  N_[1] = nCoarse;
+  for (int k = 0; k < 2; ++k) {
+    R_[k] = waferSize / (3 * N_[k]);
+    r_[k] = sqrt3By2_ * R_[k];
+  }
+}
+
+std::pair<double, double> HGCalWafer::HGCalWaferUV2XY1(int32_t u, int32_t v, int32_t placementIndex, int32_t type) {
+  if (type != 0)
+    type = 1;
+  double x(0), y(0);
+  switch (placementIndex) {
+    case (HGCalWafer::WaferPlacementIndex6):
+      x = (1.5 * (v - u) + 0.5) * R_[type];
+      y = (v + u - 2 * N_[type] + 1) * r_[type];
+      break;
+    case (HGCalWafer::WaferPlacementIndex7):
+      x = (1.5 * (v - N_[type]) + 1.0) * R_[type];
+      y = (2 * u - v - N_[type]) * r_[type];
+      break;
+    case (HGCalWafer::WaferPlacementIndex8):
+      x = (1.5 * (u - N_[type]) + 0.5) * R_[type];
+      y = -(2 * v - u - N_[type] + 1) * r_[type];
+      break;
+    case (HGCalWafer::WaferPlacementIndex9):
+      x = -(1.5 * (v - u) + 0.5) * R_[type];
+      y = -(v + u - 2 * N_[type] + 1) * r_[type];
+      break;
+    case (HGCalWafer::WaferPlacementIndex10):
+      x = -(1.5 * (v - N_[type]) + 1) * R_[type];
+      y = -(2 * u - v - N_[type]) * r_[type];
+      break;
+    case (HGCalWafer::WaferPlacementIndex11):
+      x = -(1.5 * (u - N_[type]) + 0.5) * R_[type];
+      y = (2 * v - u - N_[type] + 1) * r_[type];
+      break;
+    case (HGCalWafer::WaferPlacementIndex0):
+      x = (1.5 * (u - v) - 0.5) * R_[type];
+      y = (v + u - 2 * N_[type] + 1) * r_[type];
+      break;
+    case (HGCalWafer::WaferPlacementIndex1):
+      x = -(1.5 * (v - N_[type]) + 1.0) * R_[type];
+      y = (2 * u - v - N_[type]) * r_[type];
+      break;
+    case (HGCalWafer::WaferPlacementIndex2):
+      x = -(1.5 * (u - N_[type]) + 0.5) * R_[type];
+      y = -(2 * v - u - N_[type] + 1) * r_[type];
+      break;
+    case (HGCalWafer::WaferPlacementIndex3):
+      x = -(1.5 * (u - v) - 0.5) * R_[type];
+      y = -(v + u - 2 * N_[type] + 1) * r_[type];
+      break;
+    case (HGCalWafer::WaferPlacementIndex4):
+      x = (1.5 * (v - N_[type]) + 1) * R_[type];
+      y = -(2 * u - v - N_[type]) * r_[type];
+      break;
+    default:
+      x = (1.5 * (u - N_[type]) + 0.5) * R_[type];
+      y = (2 * v - u - N_[type] + 1) * r_[type];
+      break;
+  }
+  return std::make_pair(x, y);
+}
+
+std::pair<double, double> HGCalWafer::HGCalWaferUV2XY2(int32_t u, int32_t v, int32_t placementIndex, int32_t type) {
+  if (type != 0)
+    type = 1;
+  double x(0), y(0);
+  if (placementIndex < HGCalWafer::WaferPlacementExtra) {
+    double x0 = (1.5 * (u - v) - 0.5) * R_[type];
+    double y0 = (u + v - 2 * N_[type] + 1) * r_[type];
+    const std::vector<double> fac1 = {1.0, 0.5, -0.5, -1.0, -0.5, 0.5};
```

Comment:
Ciao @bsunanda would it make sense to rename `fac{1,2}` to something that could somehow suggest that you are performing a rotation, possibly with `sin` and `cos` inside the name?

Reply:
I am changing the names to fcos and fsin